[Ingest Pipelines] Processors editor a11y focus states

jloleysens opened the pull request:
## Summary

* Updated structure of HTML (removed unnecessary flex group and items)
* Optimised keyboard focus states in move mode
* Updated styling to better cater for overflow in certain cases

Addresses items 2 & 3 of https://github.com/elastic/kibana/issues/71666

## Test

<details>
<summary>Pipeline JSON</summary>

```
{
  "processors": [
    {
      "grok": {
        "field": "my-field",
        "patterns": [
          "enter"
        ],
        "description": "test 2"
      }
    },
    {
      "grok": {
        "field": "test",
        "patterns": [
          "test",
          "test-again"
        ],
        "description": "test 1",
        "on_failure": [
          {
            "bytes": {
              "field": "test"
            }
          }
        ]
      }
    }
  ]
}
```


</details>

1. Navigate to create pipeline processor in Ingest pipelines plugin
2. Import processors from above
3. Tab to the move button of the inner "Bytes" processor
4. Use the keyboard navigation to either cancel the move (can be cancelled with esc key too) or place the processor somewhere else. Using spacebar+tab to interact with the buttons and drop zones.

## Release note

Optimised keyboard navigation of the ingest processors component

## Gif

![a11y-focus-states-ingest-proc-editor](https://user-images.githubusercontent.com/8155004/94819261-5571d780-03ff-11eb-96c6-8cb372360c60.gif)


### Checklist

Delete any items that are not applicable to this PR.

- [x] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78754/)
* Commit: eb7d539685e25f4873f4817f2ded9255d02073db
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78754/execution/node/336/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78754/","number":"78754","commit":"eb7d539685e25f4873f4817f2ded9255d02073db"}]}
PIPELINE-->

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79084/)
* Commit: a2edd47fd36bf0086da1e3d00142bae2e1e0ecc7

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/a2edd47fd36bf0086da1e3d00142bae2e1e0ecc7) | [after](https://github.com/elastic/kibana/commit/14c7a5e41ae663a8fbf30b008b7c921d2cd24377) | diff |
| --- | --- | --- | --- |
| `ingestPipelines` | 814.4KB | 814.4KB | +4.0B |

### History
* :broken_heart: [Build #78754](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78754/) failed eb7d539685e25f4873f4817f2ded9255d02073db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78754/","number":"78754","commit":"eb7d539685e25f4873f4817f2ded9255d02073db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79084/","number":"79084","commit":"a2edd47fd36bf0086da1e3d00142bae2e1e0ecc7"}]}
PIPELINE-->

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79085/)
* Commit: a52b872b987fd15e6d1561d19ac3066d1ab7af44
* Flaky suites:
  * xpack-kibana-ciGroup9

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79085/testReport)

<details><summary>X-Pack Alerting API Integration Tests.x-pack/test/alerting_api_integration/spaces_only/tests/alerting/execution_status·ts.alerting api integration spaces only Alerting executionStatus should be able to find over all the fields</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79085/testReport//junit/X-Pack%20Alerting%20API%20Integration%20Tests/x-pack_test_alerting_api_integration_spaces_only_tests_alerting_execution_status·ts/alerting_api_integration_spaces_only_Alerting_executionStatus_should_be_able_to_find_over_all_the_fields)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 1 times on tracked branches: https://github.com/elastic/kibana/issues/79248

[00:00:00]       │
[00:00:00]         │ info [o.e.c.m.MetadataIndexTemplateService] [kibana-ci-immutable-centos-tests-xxl-1601631796344358277] adding template [.kibana-event-log-8.0.0-template] for index patterns [.kibana-event-log-8.0.0-*]
[00:00:00]         └-: alerting api integration spaces only
[00:00:00]           └-> "before all" hook
[00:00:38]           └-: Alerting
[00:00:38]             └-> "before all" hook
[00:00:38]             └-> "before all" hook
[00:00:38]               │ debg creating space
[00:00:38]               │ info [o.e.c.m.MetadataMappingService] [kibana-ci-immutable-centos-tests-xxl-1601631796344358277] [.kibana_2/pnJ87NJiTImDamnrpEzHVw] update_mapping [_doc]
[00:00:39]               │ debg created space
[00:00:39]               │ debg creating space
[00:00:40]               │ debg created space
[00:02:15]             └-: executionStatus
[00:02:15]               └-> "before all" hook
[00:02:15]               └-> should be "pending" for newly created alert
[00:02:15]                 └-> "before each" hook: global before each
[00:02:16]                 └- ✓ pass  (1.8s) "alerting api integration spaces only Alerting executionStatus should be "pending" for newly created alert"
[00:02:16]               └-> should eventually be "ok" for no-op alert
[00:02:16]                 └-> "before each" hook: global before each
[00:02:19]                 │ proc [kibana]   log   [10:45:14.946] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:14.944Z","end":"2020-10-02T10:45:14.944Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"d067d327-d4b2-4d73-9de4-1d390f840434","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.noop:d067d327-d4b2-4d73-9de4-1d390f840434: 'abc'","@timestamp":"2020-10-02T10:45:14.944Z","ecs":{"version":"1.5.0"}}
[00:02:19]                 │ proc [kibana]   log   [10:45:14.949] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:14.948Z","end":"2020-10-02T10:45:14.948Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"6fb1be08-9935-401e-b17a-b2ad8186a190","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.noop:6fb1be08-9935-401e-b17a-b2ad8186a190: 'abc'","@timestamp":"2020-10-02T10:45:14.948Z","ecs":{"version":"1.5.0"}}
[00:02:19]                 └- ✓ pass  (2.6s) "alerting api integration spaces only Alerting executionStatus should eventually be "ok" for no-op alert"
[00:02:19]               └-> should eventually be "active" for firing alert
[00:02:19]                 └-> "before each" hook: global before each
[00:02:22]                 │ proc [kibana]   log   [10:45:17.946] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:17.941Z","end":"2020-10-02T10:45:17.941Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc'","@timestamp":"2020-10-02T10:45:17.941Z","ecs":{"version":"1.5.0"}}
[00:02:22]                 │ proc [kibana]   log   [10:45:17.946] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"new-instance"},"kibana":{"alerting":{"instance_id":"instance"},"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc' created new instance: 'instance'","@timestamp":"2020-10-02T10:45:17.943Z","ecs":{"version":"1.5.0"}}
[00:02:22]                 │ proc [kibana]   log   [10:45:17.947] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"active-instance"},"kibana":{"alerting":{"instance_id":"instance"},"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc' active instance: 'instance'","@timestamp":"2020-10-02T10:45:17.944Z","ecs":{"version":"1.5.0"}}
[00:02:22]                 │ proc [kibana]   log   [10:45:17.953] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:17.951Z","end":"2020-10-02T10:45:17.951Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"6fb1be08-9935-401e-b17a-b2ad8186a190","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.noop:6fb1be08-9935-401e-b17a-b2ad8186a190: 'abc'","@timestamp":"2020-10-02T10:45:17.951Z","ecs":{"version":"1.5.0"}}
[00:02:22]                 └- ✓ pass  (3.0s) "alerting api integration spaces only Alerting executionStatus should eventually be "active" for firing alert"
[00:02:22]               └-> should eventually be "error" for an error alert
[00:02:22]                 └-> "before each" hook: global before each
[00:02:25]                 │ proc [kibana]   log   [10:45:20.944] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:20.943Z","end":"2020-10-02T10:45:20.943Z","duration":0,"outcome":"failure"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"13386da3-f69f-4545-8b0b-26415c72fee9","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert execution failure: test.throw:13386da3-f69f-4545-8b0b-26415c72fee9: 'abc'","error":{"message":"this alert is intended to fail"},"@timestamp":"2020-10-02T10:45:20.943Z","ecs":{"version":"1.5.0"}}
[00:02:25]                 │ proc [kibana]   log   [10:45:20.948] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:20.946Z","end":"2020-10-02T10:45:20.946Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc'","@timestamp":"2020-10-02T10:45:20.946Z","ecs":{"version":"1.5.0"}}
[00:02:25]                 │ proc [kibana]   log   [10:45:20.948] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"active-instance"},"kibana":{"alerting":{"instance_id":"instance"},"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc' active instance: 'instance'","@timestamp":"2020-10-02T10:45:20.947Z","ecs":{"version":"1.5.0"}}
[00:02:25]                 │ proc [kibana]   log   [10:45:20.953] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:20.952Z","end":"2020-10-02T10:45:20.952Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"6fb1be08-9935-401e-b17a-b2ad8186a190","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.noop:6fb1be08-9935-401e-b17a-b2ad8186a190: 'abc'","@timestamp":"2020-10-02T10:45:20.952Z","ecs":{"version":"1.5.0"}}
[00:02:25]                 └- ✓ pass  (3.0s) "alerting api integration spaces only Alerting executionStatus should eventually be "error" for an error alert"
[00:02:25]               └-> should eventually have error reason "execute" when appropriate
[00:02:25]                 └-> "before each" hook: global before each
[00:02:25]                 │ proc [kibana]   log   [10:45:21.835] [error][alerting][alerts][plugins][plugins] Executing Alert "13386da3-f69f-4545-8b0b-26415c72fee9" has resulted in Error: this alert is intended to fail
[00:02:28]                 │ proc [kibana]   log   [10:45:23.955] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:23.949Z","end":"2020-10-02T10:45:23.949Z","duration":0,"outcome":"failure"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"13386da3-f69f-4545-8b0b-26415c72fee9","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert execution failure: test.throw:13386da3-f69f-4545-8b0b-26415c72fee9: 'abc'","error":{"message":"this alert is intended to fail"},"@timestamp":"2020-10-02T10:45:23.949Z","ecs":{"version":"1.5.0"}}
[00:02:28]                 │ proc [kibana]   log   [10:45:23.955] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:23.951Z","end":"2020-10-02T10:45:23.951Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc'","@timestamp":"2020-10-02T10:45:23.951Z","ecs":{"version":"1.5.0"}}
[00:02:28]                 │ proc [kibana]   log   [10:45:23.956] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"active-instance"},"kibana":{"alerting":{"instance_id":"instance"},"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc' active instance: 'instance'","@timestamp":"2020-10-02T10:45:23.952Z","ecs":{"version":"1.5.0"}}
[00:02:28]                 │ proc [kibana]   log   [10:45:23.956] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:23.953Z","end":"2020-10-02T10:45:23.953Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"6fb1be08-9935-401e-b17a-b2ad8186a190","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.noop:6fb1be08-9935-401e-b17a-b2ad8186a190: 'abc'","@timestamp":"2020-10-02T10:45:23.953Z","ecs":{"version":"1.5.0"}}
[00:02:28]                 │ proc [kibana]   log   [10:45:23.962] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:23.960Z","end":"2020-10-02T10:45:23.960Z","duration":0,"outcome":"failure"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"92a499fb-9c15-4fe8-9f08-d556af480b41","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert execution failure: test.throw:92a499fb-9c15-4fe8-9f08-d556af480b41: 'abc'","error":{"message":"this alert is intended to fail"},"@timestamp":"2020-10-02T10:45:23.960Z","ecs":{"version":"1.5.0"}}
[00:02:28]                 └- ✓ pass  (3.0s) "alerting api integration spaces only Alerting executionStatus should eventually have error reason "execute" when appropriate"
[00:02:28]               └-> should eventually have error reason "unknown" when appropriate
[00:02:28]                 └-> "before each" hook: global before each
[00:02:29]                 │ proc [kibana]   log   [10:45:24.870] [error][alerting][alerts][plugins][plugins] Executing Alert "13386da3-f69f-4545-8b0b-26415c72fee9" has resulted in Error: this alert is intended to fail
[00:02:29]                 │ proc [kibana]   log   [10:45:24.873] [error][alerting][alerts][plugins][plugins] Executing Alert "92a499fb-9c15-4fe8-9f08-d556af480b41" has resulted in Error: this alert is intended to fail
[00:02:31]                 │ proc [kibana]   log   [10:45:26.948] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:26.944Z","end":"2020-10-02T10:45:26.944Z","duration":0,"outcome":"failure"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"13386da3-f69f-4545-8b0b-26415c72fee9","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert execution failure: test.throw:13386da3-f69f-4545-8b0b-26415c72fee9: 'abc'","error":{"message":"this alert is intended to fail"},"@timestamp":"2020-10-02T10:45:26.944Z","ecs":{"version":"1.5.0"}}
[00:02:31]                 │ proc [kibana]   log   [10:45:26.949] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:26.947Z","end":"2020-10-02T10:45:26.947Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"6c96def0-fe82-49e2-837d-71f7b3783a03","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.validation:6c96def0-fe82-49e2-837d-71f7b3783a03: 'abc'","@timestamp":"2020-10-02T10:45:26.947Z","ecs":{"version":"1.5.0"}}
[00:02:31]                 │ proc [kibana]   log   [10:45:26.955] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:26.951Z","end":"2020-10-02T10:45:26.951Z","duration":0,"outcome":"failure"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"92a499fb-9c15-4fe8-9f08-d556af480b41","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert execution failure: test.throw:92a499fb-9c15-4fe8-9f08-d556af480b41: 'abc'","error":{"message":"this alert is intended to fail"},"@timestamp":"2020-10-02T10:45:26.951Z","ecs":{"version":"1.5.0"}}
[00:02:31]                 │ proc [kibana]   log   [10:45:26.962] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:26.953Z","end":"2020-10-02T10:45:26.953Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc'","@timestamp":"2020-10-02T10:45:26.953Z","ecs":{"version":"1.5.0"}}
[00:02:31]                 │ proc [kibana]   log   [10:45:26.963] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"active-instance"},"kibana":{"alerting":{"instance_id":"instance"},"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc' active instance: 'instance'","@timestamp":"2020-10-02T10:45:26.953Z","ecs":{"version":"1.5.0"}}
[00:02:31]                 │ proc [kibana]   log   [10:45:26.969] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:26.968Z","end":"2020-10-02T10:45:26.968Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"6fb1be08-9935-401e-b17a-b2ad8186a190","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.noop:6fb1be08-9935-401e-b17a-b2ad8186a190: 'abc'","@timestamp":"2020-10-02T10:45:26.968Z","ecs":{"version":"1.5.0"}}
[00:02:32]                 │ proc [kibana]   log   [10:45:27.899] [error][alerting][alerts][plugins][plugins] Executing Alert "13386da3-f69f-4545-8b0b-26415c72fee9" has resulted in Error: this alert is intended to fail
[00:02:32]                 │ proc [kibana]   log   [10:45:27.902] [error][alerting][alerts][plugins][plugins] Executing Alert "92a499fb-9c15-4fe8-9f08-d556af480b41" has resulted in Error: this alert is intended to fail
[00:02:34]                 │ proc [kibana]   log   [10:45:29.953] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:29.949Z","end":"2020-10-02T10:45:29.949Z","duration":0,"outcome":"failure"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"92a499fb-9c15-4fe8-9f08-d556af480b41","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert execution failure: test.throw:92a499fb-9c15-4fe8-9f08-d556af480b41: 'abc'","error":{"message":"this alert is intended to fail"},"@timestamp":"2020-10-02T10:45:29.949Z","ecs":{"version":"1.5.0"}}
[00:02:34]                 │ proc [kibana]   log   [10:45:29.960] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:29.955Z","end":"2020-10-02T10:45:29.955Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc'","@timestamp":"2020-10-02T10:45:29.955Z","ecs":{"version":"1.5.0"}}
[00:02:34]                 │ proc [kibana]   log   [10:45:29.961] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"active-instance"},"kibana":{"alerting":{"instance_id":"instance"},"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc' active instance: 'instance'","@timestamp":"2020-10-02T10:45:29.956Z","ecs":{"version":"1.5.0"}}
[00:02:34]                 │ proc [kibana]   log   [10:45:29.961] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:29.957Z","end":"2020-10-02T10:45:29.957Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"6fb1be08-9935-401e-b17a-b2ad8186a190","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.noop:6fb1be08-9935-401e-b17a-b2ad8186a190: 'abc'","@timestamp":"2020-10-02T10:45:29.957Z","ecs":{"version":"1.5.0"}}
[00:02:34]                 │ proc [kibana]   log   [10:45:29.961] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:29.959Z","end":"2020-10-02T10:45:29.959Z","duration":0,"outcome":"failure"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"13386da3-f69f-4545-8b0b-26415c72fee9","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert execution failure: test.throw:13386da3-f69f-4545-8b0b-26415c72fee9: 'abc'","error":{"message":"this alert is intended to fail"},"@timestamp":"2020-10-02T10:45:29.959Z","ecs":{"version":"1.5.0"}}
[00:02:34]                 └- ✓ pass  (5.6s) "alerting api integration spaces only Alerting executionStatus should eventually have error reason "unknown" when appropriate"
[00:02:34]               └-> should be able to find over all the fields
[00:02:34]                 └-> "before each" hook: global before each
[00:02:35]                 │ proc [kibana]   log   [10:45:30.915] [error][alerting][alerts][plugins][plugins] Executing Alert "92a499fb-9c15-4fe8-9f08-d556af480b41" has resulted in Error: this alert is intended to fail
[00:02:35]                 │ proc [kibana]   log   [10:45:30.916] [error][alerting][alerts][plugins][plugins] Executing Alert "6c96def0-fe82-49e2-837d-71f7b3783a03" has resulted in Error: params invalid: [param1]: expected value of type [string] but got [number]
[00:02:35]                 │ proc [kibana]   log   [10:45:30.918] [error][alerting][alerts][plugins][plugins] Executing Alert "13386da3-f69f-4545-8b0b-26415c72fee9" has resulted in Error: this alert is intended to fail
[00:02:37]                 │ proc [kibana]   log   [10:45:32.962] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:32.955Z","end":"2020-10-02T10:45:32.955Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc'","@timestamp":"2020-10-02T10:45:32.955Z","ecs":{"version":"1.5.0"}}
[00:02:37]                 │ proc [kibana]   log   [10:45:32.962] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"active-instance"},"kibana":{"alerting":{"instance_id":"instance"},"saved_objects":[{"rel":"primary","type":"alert","id":"da65e9c1-e106-4937-8a25-f333e8060ff0","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"test.patternFiring:da65e9c1-e106-4937-8a25-f333e8060ff0: 'abc' active instance: 'instance'","@timestamp":"2020-10-02T10:45:32.956Z","ecs":{"version":"1.5.0"}}
[00:02:37]                 │ proc [kibana]   log   [10:45:32.963] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:32.957Z","end":"2020-10-02T10:45:32.958Z","duration":1000000,"outcome":"failure"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"92a499fb-9c15-4fe8-9f08-d556af480b41","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert execution failure: test.throw:92a499fb-9c15-4fe8-9f08-d556af480b41: 'abc'","error":{"message":"this alert is intended to fail"},"@timestamp":"2020-10-02T10:45:32.958Z","ecs":{"version":"1.5.0"}}
[00:02:37]                 │ proc [kibana]   log   [10:45:32.963] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:32.959Z","end":"2020-10-02T10:45:32.959Z","duration":0,"outcome":"failure"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"13386da3-f69f-4545-8b0b-26415c72fee9","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert execution failure: test.throw:13386da3-f69f-4545-8b0b-26415c72fee9: 'abc'","error":{"message":"this alert is intended to fail"},"@timestamp":"2020-10-02T10:45:32.959Z","ecs":{"version":"1.5.0"}}
[00:02:37]                 │ proc [kibana]   log   [10:45:32.967] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:32.966Z","end":"2020-10-02T10:45:32.966Z","duration":0,"outcome":"failure"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"875dfa80-e67f-4a94-896c-719429179a5a","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert execution failure: test.throw:875dfa80-e67f-4a94-896c-719429179a5a: 'abc'","error":{"message":"this alert is intended to fail"},"@timestamp":"2020-10-02T10:45:32.966Z","ecs":{"version":"1.5.0"}}
[00:02:37]                 │ proc [kibana]   log   [10:45:32.974] [info][eventLog][plugins] event logged: {"event":{"provider":"alerting","action":"execute","start":"2020-10-02T10:45:32.973Z","end":"2020-10-02T10:45:32.973Z","duration":0,"outcome":"success"},"kibana":{"saved_objects":[{"rel":"primary","type":"alert","id":"6fb1be08-9935-401e-b17a-b2ad8186a190","namespace":"space1"}],"server_uuid":"5b2de169-2785-441b-ae8c-186a1936b17d"},"message":"alert executed: test.noop:6fb1be08-9935-401e-b17a-b2ad8186a190: 'abc'","@timestamp":"2020-10-02T10:45:32.973Z","ecs":{"version":"1.5.0"}}
[00:02:37]                 └- ✖ fail: alerting api integration spaces only Alerting executionStatus should be able to find over all the fields
[00:02:37]                 │      TypeError: Cannot destructure property `executionStatus` of 'undefined' or 'null'.
[00:02:37]                 │       at waitForFindStatus (test/alerting_api_integration/spaces_only/tests/alerting/execution_status.ts:293:52)
[00:02:37]                 │ 
[00:02:37]                 │ 

```

#### Stack Trace
```
TypeError: Cannot destructure property `executionStatus` of 'undefined' or 'null'.
    at waitForFindStatus (test/alerting_api_integration/spaces_only/tests/alerting/execution_status.ts:293:52)
```
</details>

---

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/a52b872b987fd15e6d1561d19ac3066d1ab7af44) | [after](https://github.com/elastic/kibana/commit/e52884cfa285ee0aeae6ba1ad7655efefddb67c3) | diff |
| --- | --- | --- | --- |
| `ingestPipelines` | 814.3KB | 814.3KB | +4.0B |

### History
* :green_heart: [Build #79084](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79084/) succeeded a2edd47fd36bf0086da1e3d00142bae2e1e0ecc7
* :broken_heart: [Build #78754](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78754/) failed eb7d539685e25f4873f4817f2ded9255d02073db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78754/","number":"78754","commit":"eb7d539685e25f4873f4817f2ded9255d02073db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79084/","number":"79084","commit":"a2edd47fd36bf0086da1e3d00142bae2e1e0ecc7"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79085/","number":"79085","commit":"a52b872b987fd15e6d1561d19ac3066d1ab7af44"}]}
PIPELINE-->

alisonelizabeth reviewed (approved):
Changes LGTM. Thanks for making these improvements @jloleysens!

cchaos reviewed (commented):
Adding and removing `tabindex` properties from  semantically both interact-able (button) and non-interact-able (div) seems like a big accessibility issue. Has this been tested with screen-readers?

jloleysens commented:
Hi @cchaos ! In this issue https://github.com/elastic/kibana/issues/71666 @myasonik and I went through this component with a screen reader and identified this as something that we can optimise for navigating with keyboard.

This PR specifically addresses points 2 & 3 of that issue. Turning tabindex off for certain otherwise tabbable components enables faster traversal with a keyboard to the "drop zones".

Does this answer your question or was there a specific concern that I hadn't considered in the proposed changes?

cchaos reviewed (approved):
Thanks for the history and explanation! So long as there's been discussion and thought towards a11y 👍

myasonik commented:
Hey, sorry to hold this up a touch, but I want to give this a pass through. Will try to get to it before end of day.

jloleysens commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79480/)
* Commit: 9c5c5f650eda8248a5b237a327f41f6d646565c4

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### async chunks size
| id | [before](https://github.com/elastic/kibana/commit/9c5c5f650eda8248a5b237a327f41f6d646565c4) | [after](https://github.com/elastic/kibana/commit/9e1c44fcc08351070afffe87056cd86aefbe3449) | diff |
| --- | --- | --- | --- |
| `ingestPipelines` | 812.9KB | 812.9KB | +4.0B |

### History
* :yellow_heart: [Build #79085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79085/) was flaky a52b872b987fd15e6d1561d19ac3066d1ab7af44
* :green_heart: [Build #79084](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79084/) succeeded a2edd47fd36bf0086da1e3d00142bae2e1e0ecc7
* :broken_heart: [Build #78754](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78754/) failed eb7d539685e25f4873f4817f2ded9255d02073db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/78754/","number":"78754","commit":"eb7d539685e25f4873f4817f2ded9255d02073db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79084/","number":"79084","commit":"a2edd47fd36bf0086da1e3d00142bae2e1e0ecc7"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79085/","number":"79085","commit":"a52b872b987fd15e6d1561d19ac3066d1ab7af44"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/79480/","number":"79480","commit":"9c5c5f650eda8248a5b237a327f41f6d646565c4"}]}
PIPELINE-->

myasonik reviewed (approved):
👍 This doesn't fix all the a11y issues on the page but it does improve the state of things for keyboard users so it's best to merge

For a little more context on the screen reader front, a lot of problems on the page stem from using `react-virtualized` so the situation won't really be improved there until we can swap to using `react-window`
